worlds/stardew_valley/data/mod_logic.py

```diff
@@ -0,0 +1,70 @@
+from typing import List
+from .region_data import SVRegion
+from .villagers_data import all_villagers
+from ..stardew_rule import False_
+from .. import options
+from ..options import StardewOptions
+
+# Attempt to inject new methods into logic that can be easily commented out.
+
+
+def append_mod_skill_level(skills_items: List[str], world_options: StardewOptions):
+    if "Luck Skill" in world_options[options.Mods]:
+        skills_items.append("Luck Level")
+    if "Socializing Skill" in world_options[options.Mods]:
+        skills_items.append("Socializing Level")
+    if "Magic" in world_options[options.Mods]:
+        skills_items.append("Magic Level")
+    if "Archaeology" in world_options[options.Mods]:
+        skills_items.append("Archaeology Level")
+    if "Binning Skill" in world_options[options.Mods]:
+        skills_items.append("Binning Level")
+    if "Cooking Skill" in world_options[options.Mods]:
+        skills_items.append("Cooking Level")
+
+
+def can_earn_mod_skill_level(rule, skill: str, level: int, tool_rule: bool):
+    mod_tool_rule = tool_rule
+    if "Luck Skill" in rule.options[options.Mods]:
+        if skill == "Luck":
+            if level >= 6:
+                mod_tool_rule = rule.can_fish_chests() | rule.can_open_geode("Magma Geode")
+            else:
+                mod_tool_rule = rule.can_fish_chests() | rule.can_open_geode("Geode")
+    if "Magic" in rule.options[options.Mods]:
+        if skill == "Magic":
+            if level >= 6:
+                mod_tool_rule = rule.can_earn_spell_count(10)
+            else:
+                mod_tool_rule = rule.can_earn_spell_count(5)
+    if "Socializing Skill" in rule.options[options.Mods]:
+        if skill == "Socializing":
+            villager_count = 0
+            for villager in all_villagers:
+                if villager.mod_name in rule.options[options.Mods]:
+                    if rule.can_earn_relationship(villager.name, level):
+                        villager_count += 1
+            if villager_count < level * 2:
+                mod_tool_rule = False_()
+    if "Archaeology" in rule.options[options.Mods]:
+        if skill == "Archaeology":
+            if level >= 6:
+                mod_tool_rule = rule.can_do_panning() | rule.can_mine_in_the_mines_floor_81_120()
+            else:
+                mod_tool_rule = rule.can_do_panning() | rule.can_mine_in_the_mines_floor_1_40()
```

Comment:
Archaeology is a skill tree which expands on finding artifacts and minerals, which includes making panning more worthwhile, and panning gives experience.  Digging up artifacts also counts and includes the mines.  The rule could be shifted to reaching every town region where artifact spots show up?  There's more here:

https://www.nexusmods.com/stardewvalley/mods/15793

Reply:
since there are artifact spots on pretty much every map, I'd say that owning a hoe should be sufficient for that. Maybe lock it behind hoe levels to create a bit of artificial progression (2 levels of skill per level of hoe?).